src/context/framework.provider.tsx

```diff
@@ -0,0 +1,67 @@
+import { ComponentType, createContext, ReactNode, useContext, useMemo } from "react";
+
+export enum Framework {
+  PPC = "ppc",
+  TF = "terrafund",
+  TF_LANDSCAPES = "terrafund-landscapes",
+  ENTERPRISES = "enterprises",
+  HBF = "hbf",
+
+  UNDEFINED = "undefined"
+}
+
+export const ALL_TF = [Framework.TF, Framework.TF_LANDSCAPES, Framework.ENTERPRISES];
+
+interface IFrameworkContext {
+  framework: Framework;
+}
+
+export const FrameworkContext = createContext<IFrameworkContext>({
+  framework: Framework.UNDEFINED
+});
+
+type FrameworkProviderProps = { children: ReactNode; frameworkKey?: string };
+
+const FrameworkProvider = ({ children, frameworkKey }: FrameworkProviderProps) => {
+  const framework = useMemo(
+    () =>
+      Object.values(Framework).includes(frameworkKey as unknown as Framework)
+        ? (frameworkKey as Framework)
+        : Framework.UNDEFINED,
+    [frameworkKey]
+  );
+
+  return <FrameworkContext.Provider value={{ framework }}>{children}</FrameworkContext.Provider>;
+};
+
+export const useFrameworkContext = () => useContext(FrameworkContext);
+
+export interface ShowHideProps {
+  // The element will only be shown if the current framework is in this list.
+  frameworksShow?: Framework[];
+  // The element will only be shown if the current framework is not in this list. `hide` will be
+  // ignored if `show` is also included.
+  frameworksHide?: Framework[];
+}
+
+export const useFrameworkShowHide = ({ frameworksShow, frameworksHide }: ShowHideProps) => {
+  const { framework } = useFrameworkContext();
+
+  if (frameworksShow != null) return frameworksShow.includes(framework);
+  if (frameworksHide != null) return !frameworksHide.includes(framework);
+  return true;
+};
+
+export function withFrameworkShow<T>(WrappedComponent: ComponentType<T>) {
+  const displayName = WrappedComponent.displayName ?? WrappedComponent.name ?? "Component";
+  const FrameworkShowHide = (props: T & ShowHideProps) => {
+    const { frameworksShow, frameworksHide, ...rest } = props;
+    if (!useFrameworkShowHide({ frameworksShow, frameworksHide })) return null;
+
+    return <WrappedComponent {...(rest as T & JSX.IntrinsicAttributes)} />;
+  };
+  FrameworkShowHide.displayName = `withFrameworkShow(${displayName})`;
+  return FrameworkShowHide;
+}
+
+export default FrameworkProvider;
```

Comment:
This file is the main workhorse of the new system. It provides:
* A new react context that provides the current Framework (as defined in the enum in this file) to components in the render tree
* a hook (`useFrameworkContext`) to get the current contextual framework.
* an HOC (`withFrameworkShow`) to wrap a component with the logic to show or hide that component based on the current framework.